common/utils.go

```diff
@@ -71,6 +71,36 @@ type metadataWrapper struct {
 	Metadata Metadata `json:"compute"`
 }
 
+var (
+	// Creating http client object to be reused instead of creating one every time.
+	// This helps make use of the cached tcp connections.
+	// Clients are safe for concurrent use by multiple goroutines.
+	httpClient *http.Client
+)
+
+// InitHttpClient initializes the httpClient object
+func InitHttpClient(
+	connectionTimeoutSec int,
+	responseHeaderTimeoutSec int) *http.Client {
+	log.Printf("[Utils] Initializing HTTP client with connection timeout: %d, response header timeout: %d",
+		connectionTimeoutSec, responseHeaderTimeoutSec)
+	httpClient = &http.Client{
+		Transport: &http.Transport{
+			DialContext: (&net.Dialer{
+				Timeout: time.Duration(connectionTimeoutSec) * time.Second,
```

Comment:
can the intialization fail? are we supposed to check if httpclient is null or not?

Reply:
The init cannot fail. httpClient is created when CNS starts, so it will be available